From 52ef6101e0badfb2ab8838812e9936a08961156a Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Tue, 17 Nov 2015 09:17:37 -0600
Subject: [PATCH 111/213] drivers/misc/lsi-ncr.c: Move the Header to the
 Default Path

Move lsi-ncr.h to include/linux as it is used in arch/arm/* and
arch/powerpc/* among other places.  All source files including
lsi-ncr.h are updated as well.

Signed-off-by: John Jacques <john.jacques@intel.com>
---
 arch/arm/mach-axxia/axxia.c               |  2 +-
 arch/arm/mach-axxia/ddr_retention.c       |  4 +--
 arch/arm/mach-axxia/perf_event_platform.c |  2 +-
 arch/arm/mach-axxia/rapidio.c             |  2 +-
 arch/arm/mach-axxia/smon.c                |  3 ++-
 drivers/edac/axxia_edac-l2_cpu.c          |  2 +-
 drivers/edac/axxia_edac-l3.c              |  2 +-
 drivers/edac/axxia_edac-mc.c              |  2 +-
 drivers/misc/lsi-ncr.c                    |  2 +-
 drivers/misc/lsi-ncr.h                    | 43 -------------------------------
 drivers/net/ethernet/lsi/lsi_acp_net.c    |  3 +--
 include/linux/lsi-ncr.h                   | 43 +++++++++++++++++++++++++++++++
 12 files changed, 55 insertions(+), 55 deletions(-)
 delete mode 100644 drivers/misc/lsi-ncr.h
 create mode 100644 include/linux/lsi-ncr.h

diff --git a/arch/arm/mach-axxia/axxia.c b/arch/arm/mach-axxia/axxia.c
index 1cccc8f..c628bf2 100644
--- a/arch/arm/mach-axxia/axxia.c
+++ b/arch/arm/mach-axxia/axxia.c
@@ -49,7 +49,7 @@
 #include <mach/timers.h>
 #include <mach/axxia-gic.h>
 #include <linux/irqchip/arm-gic.h>
-#include "../../drivers/misc/lsi-ncr.h"
+#include <linux/lsi-ncr.h>
 #include "axxia.h"
 #include "pci.h"
 #ifdef CONFIG_AXXIA_RIO
diff --git a/arch/arm/mach-axxia/ddr_retention.c b/arch/arm/mach-axxia/ddr_retention.c
index 989269a..4649832 100644
--- a/arch/arm/mach-axxia/ddr_retention.c
+++ b/arch/arm/mach-axxia/ddr_retention.c
@@ -25,12 +25,12 @@
 #include <linux/proc_fs.h>
 #include <linux/prefetch.h>
 #include <linux/delay.h>
-
 #include <linux/of.h>
+#include <linux/lsi-ncr.h>
+
 #include <asm/io.h>
 #include <asm/cacheflush.h>
 #include "axxia.h"
-#include "../../drivers/misc/lsi-ncr.h"
 
 static void __iomem *nca;
 static void __iomem *apb;
diff --git a/arch/arm/mach-axxia/perf_event_platform.c b/arch/arm/mach-axxia/perf_event_platform.c
index e2ad209..85bef3a 100644
--- a/arch/arm/mach-axxia/perf_event_platform.c
+++ b/arch/arm/mach-axxia/perf_event_platform.c
@@ -45,8 +45,8 @@
 #include <linux/proc_fs.h>
 
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 #include <asm/cacheflush.h>
-#include <../../../drivers/misc/lsi-ncr.h>
 
 #include "perf_event_platform.h"
 
diff --git a/arch/arm/mach-axxia/rapidio.c b/arch/arm/mach-axxia/rapidio.c
index 3d95a43..d3e85df 100644
--- a/arch/arm/mach-axxia/rapidio.c
+++ b/arch/arm/mach-axxia/rapidio.c
@@ -27,9 +27,9 @@
 #include <linux/clk.h>
 #include <linux/delay.h>
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 #include <linux/signal.h>
 
-#include "../../drivers/misc/lsi-ncr.h"
 #include <mach/rio.h>
 
 /**
diff --git a/arch/arm/mach-axxia/smon.c b/arch/arm/mach-axxia/smon.c
index ac1a6a1..17a1cfb 100644
--- a/arch/arm/mach-axxia/smon.c
+++ b/arch/arm/mach-axxia/smon.c
@@ -17,9 +17,10 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.
  */
+
 #include <linux/io.h>
 
-#include <../../../drivers/misc/lsi-ncr.h>
+#include <linux/lsi-ncr.h>
 
 #include "smon.h"
 
diff --git a/drivers/edac/axxia_edac-l2_cpu.c b/drivers/edac/axxia_edac-l2_cpu.c
index a0ec730..3c87f4c 100644
--- a/drivers/edac/axxia_edac-l2_cpu.c
+++ b/drivers/edac/axxia_edac-l2_cpu.c
@@ -24,8 +24,8 @@
 #include <linux/init.h>
 #include <linux/slab.h>
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 #include <linux/edac.h>
-#include "../misc/lsi-ncr.h"
 #include <linux/of_platform.h>
 #include <linux/of.h>
 #include <linux/of_address.h>
diff --git a/drivers/edac/axxia_edac-l3.c b/drivers/edac/axxia_edac-l3.c
index 8d124fa..fe62107 100644
--- a/drivers/edac/axxia_edac-l3.c
+++ b/drivers/edac/axxia_edac-l3.c
@@ -24,8 +24,8 @@
 #include <linux/init.h>
 #include <linux/slab.h>
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 #include <linux/edac.h>
-#include "../misc/lsi-ncr.h"
 #include <linux/of_platform.h>
 #include <linux/of.h>
 #include <linux/of_address.h>
diff --git a/drivers/edac/axxia_edac-mc.c b/drivers/edac/axxia_edac-mc.c
index 863a692..8b5989a 100644
--- a/drivers/edac/axxia_edac-mc.c
+++ b/drivers/edac/axxia_edac-mc.c
@@ -24,8 +24,8 @@
 #include <linux/init.h>
 #include <linux/slab.h>
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 #include <linux/edac.h>
-#include "../misc/lsi-ncr.h"
 #include <linux/of_platform.h>
 #include <linux/of.h>
 #include <linux/of_address.h>
diff --git a/drivers/misc/lsi-ncr.c b/drivers/misc/lsi-ncr.c
index 9482a5f..debbfd9 100644
--- a/drivers/misc/lsi-ncr.c
+++ b/drivers/misc/lsi-ncr.c
@@ -20,8 +20,8 @@
 
 #include <linux/module.h>
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 #include <linux/of.h>
-#include "lsi-ncr.h"
 
 #ifdef CONFIG_ARCH_AXXIA
 #define NCA_PHYS_ADDRESS 0x002020100000ULL
diff --git a/drivers/misc/lsi-ncr.h b/drivers/misc/lsi-ncr.h
deleted file mode 100644
index 687dbae..0000000
--- a/drivers/misc/lsi-ncr.h
+++ /dev/null
@@ -1,43 +0,0 @@
-/*
- * drivers/lsi/common/version.h
- *
- * Copyright (C) 2010 LSI
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.
- */
-
-#ifndef __DRIVERS_LSI_ACP_NCR_H
-#define __DRIVERS_LSI_ACP_NCR_H
-
-#ifndef NCP_REGION_ID
-#define NCP_REGION_ID(node, target) \
-(unsigned long) ((((node) & 0xffff) << 16) | ((target) & 0xffff))
-#endif
-
-#ifndef NCP_NODE_ID
-#define NCP_NODE_ID(region) (((region) >> 16) & 0xffff)
-#endif
-
-#ifndef NCP_TARGET_ID
-#define NCP_TARGET_ID(region) ((region) & 0xffff)
-#endif
-
-unsigned long ncr_register_read(unsigned *);
-void ncr_register_write(const unsigned, unsigned *);
-int ncr_read(unsigned long, unsigned long, int, void *);
-int ncr_write(unsigned long, unsigned long, int, void *);
-int ncr_read_nolock(unsigned long, unsigned long, int, void *);
-int ncr_write_nolock(unsigned long, unsigned long, int, void *);
-
-#endif /*  __DRIVERS_LSI_ACP_NCR_H */
diff --git a/drivers/net/ethernet/lsi/lsi_acp_net.c b/drivers/net/ethernet/lsi/lsi_acp_net.c
index fb0389f..f431e34 100644
--- a/drivers/net/ethernet/lsi/lsi_acp_net.c
+++ b/drivers/net/ethernet/lsi/lsi_acp_net.c
@@ -75,11 +75,10 @@
 #include <linux/dma-mapping.h>
 #include <linux/uaccess.h>
 #include <linux/io.h>
+#include <linux/lsi-ncr.h>
 
 #include <asm/dma.h>
 
-#include "../../../misc/lsi-ncr.h"
-
 #include "lsi_acp_net.h"
 
 #define LSI_DRV_NAME           "acp-femac"
diff --git a/include/linux/lsi-ncr.h b/include/linux/lsi-ncr.h
new file mode 100644
index 0000000..e9c179d
--- /dev/null
+++ b/include/linux/lsi-ncr.h
@@ -0,0 +1,43 @@
+/*
+ * drivers/lsi/common/version.h
+ *
+ * Copyright (C) 2010 LSI
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.
+ */
+
+#ifndef __DRIVERS_LSI_ACP_NCR_H
+#define __DRIVERS_LSI_ACP_NCR_H
+
+#ifndef NCP_REGION_ID
+#define NCP_REGION_ID(node, target) \
+((unsigned long) ((((node) & 0xffff) << 16) | ((target) & 0xffff)))
+#endif
+
+#ifndef NCP_NODE_ID
+#define NCP_NODE_ID(region) (((region) >> 16) & 0xffff)
+#endif
+
+#ifndef NCP_TARGET_ID
+#define NCP_TARGET_ID(region) ((region) & 0xffff)
+#endif
+
+unsigned long ncr_register_read(unsigned *);
+void ncr_register_write(const unsigned, unsigned *);
+int ncr_read(unsigned long, unsigned long, int, void *);
+int ncr_write(unsigned long, unsigned long, int, void *);
+int ncr_read_nolock(unsigned long, unsigned long, int, void *);
+int ncr_write_nolock(unsigned long, unsigned long, int, void *);
+
+#endif /*  __DRIVERS_LSI_ACP_NCR_H */
-- 
1.9.1

